webui/react/src/components/ResourceAllocationModal.tsx

```diff
@@ -0,0 +1,131 @@
+import Button from 'hew/Button';
+import Link from 'hew/Link';
+import { Modal } from 'hew/Modal';
+import { Loadable } from 'hew/utils/loadable';
+import { useObservable } from 'micro-observables';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import NodeElement, { SimplifiedNode } from 'pages/ResourcePool/NodeElement';
+import { handlePath, paths } from 'routes/utils';
+import clusterStore from 'stores/cluster';
+import { Agent, Resource } from 'types';
+import { AnyMouseEvent } from 'utils/routes';
+
+import css from './ResourceAllocationModal.module.scss';
+
+export interface Node {
+  nodeName: string;
+  slotsIds: Resource[] | SimplifiedNode[];
+}
+
+export interface Props {
+  rpName: string;
+  accelerator: string;
+  nodes: Node[];
+  onClose: () => void;
+  isRunning: boolean;
+}
+
+interface InfoContainerProps {
+  label: string;
+  info?: JSX.Element | string;
+}
+
+const InfoContainer: React.FC<InfoContainerProps> = ({ label, info }) => {
+  return (
+    <div className={css.infoContainer}>
+      <span>{label}</span>
+      <span className={css.dots} />
+      <span>{info}</span>
+    </div>
+  );
+};
+
+const ResourceAllocationModalComponent: React.FC<Props> = ({
+  rpName,
+  nodes,
+  accelerator,
+  isRunning,
+  onClose,
+}: Props) => {
+  const [limitSlots, setLimitSlots] = useState(0);
+  const [activeNodes, setActiveNodes] = useState<Agent[]>([]);
+  const agents = Loadable.waitFor(useObservable(clusterStore.agents));
+  const footer = (
+    <div className={css.footer}>
+      <Button onClick={onClose}>Done</Button>
+    </div>
+  );
+  const renderNodes = useMemo(() => {
+    if (activeNodes.length > 0) {
+      return activeNodes.map(({ id, resources, slots }) => (
+        <NodeElement
+          key={id}
```

Comment:
bug: not sure which key the warning referring to, but i got many warnings like the following. 

```
Warning: Encountered two children with the same key, `info_det-agent-release-party-dear-goldfish-00001`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
```